src/transformers/models/vibevoice/generation_vibevoice.py

```diff
@@ -0,0 +1,765 @@
+                if streamer is not None:
+                    streamer.put(audio_chunk, diffusion_indices)
+
+                # Get semantic features for next step
+                semantic_outputs = self.semantic_tokenizer.encode(
+                    audio_chunk,
+                    padding_cache=semantic_cache,
+                    use_cache=self.config.use_cache,
+                )
+                semantic_features = semantic_outputs.latents[diffusion_indices]
+                semantic_cache = semantic_outputs.padding_cache
+
+                # Combine features for next input
+                acoustic_embed = self.acoustic_connector(speech_latent)
+                semantic_embed = self.semantic_connector(semantic_features)
+                diffusion_embeds = acoustic_embed + semantic_embed
+                next_inputs_embeds[diffusion_indices] = diffusion_embeds
+
+            inputs_embeds = next_inputs_embeds
+            unfinished_sequences = unfinished_sequences & ~finished_tags.long()
+            step += 1
+            # ============================================
+
+            if streamer is not None:
+                streamer.put(next_tokens.cpu())
+
+            unfinished_sequences = unfinished_sequences & ~stopping_criteria(input_ids, scores)
+            this_peer_finished = unfinished_sequences.max() == 0
+            cur_len += 1
+
+            # This is needed to properly delete outputs.logits which may be very large for first iteration
+            # Otherwise a reference to outputs is kept which keeps the logits alive in the next iteration
+            del outputs
+
+            # *************** VibeVoice specific ***************
+            del negative_outputs
+            # ============================================
+
+        if streamer is not None:
+            streamer.end()
+
+        # *************** VibeVoice specific ***************
+        # Prepare final audio outputs
+        final_audio_outputs = []
+        for sample_chunks in audio_chunks:
+            if sample_chunks:
+                final_audio_outputs.append(torch.cat(sample_chunks, dim=-1))
+            else:
+                final_audio_outputs.append(None)
+
+        if return_dict_in_generate:
+            return VibeVoiceGenerateOutput(
+                sequences=input_ids,
+                scores=scores,
+                logits=raw_logits,
+                attentions=decoder_attentions,
+                hidden_states=decoder_hidden_states,
+                past_key_values=model_kwargs.get("past_key_values"),
+                audio=final_audio_outputs,
+                reach_max_step_sample=completion_steps >= max_step_per_sample,
+            )
+        else:
+            # NOTE (ebezzam): new tokens in input_ids are simply speech tokens (mainly `speech_diffusion_id`)
+            # so returning `input_ids` is insufficient for generated audio
+            return final_audio_outputs
```

Comment:
Linked to above (_difference with other models that use tokenizers_)

I directly return the audio because the generated token IDs only have a repeated `speech_diffusion_id`. Namely they aren't token IDs that can be decoded for audio, but rather "flags" to the generation process to continue generating speech for a particular sample in the batch (until `speech_end_id` is reached)